lib/contentTypeParser.js

```diff
@@ -117,7 +138,44 @@ ContentTypeParser.prototype.run = function (contentType, handler, request, reply
   }
 }
 
-function rawBody (request, reply, options, parser, done) {
+function onPredecodingHookRunner (functions, request, reply, payload, cb) {
+  let i = 0
+
+  function next (err, newPayload) {
+    if (reply.sent) {
+      return
+    }
+
+    if (err) {
+      reply.code(err.statusCode || 500).send(err)
+      return
+    }
+
+    if (i === functions.length) {
+      cb(null, newPayload)
+      return
+    }
+
+    const fn = functions[i++]
+    const result = fn(request, reply, newPayload, next)
+
+    if (result && typeof result.then === 'function') {
+      result.then(handleResolve, handleReject)
+    }
+  }
+
+  function handleResolve (newRaw) {
+    next(null, newRaw)
+  }
+
+  function handleReject (err) {
```

Comment:
It happened that we received an empty error, we managed like this:

```
if (!err) {
      err = new FST_ERR_SEND_UNDEFINED_ERR()
    } else if (!(err instanceof Error)) {
      reply[kReplyIsError] = true
    }
```

Reply:
Ok, will do. Thanks!